### DISCUSSION TITLE
bRPC 源码分析「三、网络通信」 | SF-Zhou's Blog

### DISCUSSION THREAD

**Discussion by SF-Zhou** [SF-Zhou/sf-zhou.github.io#159] [Announcements]
https://sf-zhou.github.io/brpc/brpc_03_socket.html

**Comment by PengBY91** [comment]
赞👍，要是代码简短一点就更好了，阅读体验会好很多～

**Comment by PengBY91** [comment]
可以使用折叠

<details>
  <summary>这是折叠测试</summary>
<pre>
if (_versioned_ref.compare_exchange_weak(
      vref, MakeVRef(id_ver, nref + 1/*note*/),
      butil::memory_order_release,
      butil::memory_order_relaxed)) {
      // Set this flag to true since we add additional ref again
      // 清理 recycle 标记
      _recycle_flag.store(false, butil::memory_order_relaxed);
      if (_user) {
        // 调用 AfterRevived，默认会打印 Revived ... (Connectable)
        _user->AfterRevived(this);
      } else {
        LOG(INFO) << "Revived " << *this << " (Connectable)";
      }
      return;
    }
  </pre>
</details>

**Comment by SF-Zhou** [comment]
> @cupwater
> 赞👍，要是代码简短一点就更好了，阅读体验会好很多～

我翻了一下我之前写的，都是这种风格，估计不改了。官方文档上文字性的描述会更多一些，我写的会放更多代码细节。

**Comment by weidong-su** [comment]
求问下 画流程图的软件是什么呀

**Reply by SF-Zhou** [reply]
draw.io，VS Code 里有插件
